### PR TITLE
docs: cleaned up services page

### PR DESCRIPTION
**Pull request by MarcusNoble**
Issue #, if available: N/A

Description of changes:

Added / fixed in-page links. Removed duplicate EFS entry. Added links to relevant issue on GitHub.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/contents/services.md [hunk 1]
```diff
@@ -125,12 +125,6 @@ Controller Release Roadmap](https://github.com/aws/aws-controllers-k8s/projects/
 * Next milestone: [`DEVELOPER PREVIEW`](https://github.com/aws/aws-controllers-k8s/milestone/10)
 * AWS service documentation: https://aws.amazon.com/lambda/
 
-## Amazon EFS
-
-* Proposed: https://github.com/aws/aws-controllers-k8s/issues/328
-* Current release status: `PROPOSED`
-* AWS service documentation: https://aws.amazon.com/efs/
-
```
Comment (jaypipes) [inline docs/contents/services.md:133]: Any reason to remove the above? It's still listed in the table above...

**Review by jaypipes — COMMENTED** [review]
Thank you @MarcusNoble! Left one question inline for you...

File: docs/contents/services.md (hunk 1 above)
Comment (MarcusNoble) [reply docs/contents/services.md:133]: It was duplicated: https://github.com/aws/aws-controllers-k8s/pull/671/files#diff-d34969a303cc6cb28f7d748f100fc323faa0f65400880d06b8e3f4e902c5d268R83

**Review by jaypipes — APPROVED** [review]
thanks @MarcusNoble sorry for the delay!

**Pull request merged**